packages/blade/src/components/Accordion/AccordionButton.native.tsx

```diff
@@ -1,22 +1,136 @@
-  // const { showNumberPrefix, expandedIndex } = useAccordion();
+const _AccordionButton = ({ index, icon: Icon, children }: AccordionButtonProps): ReactElement => {
+  const { onExpandChange, isExpanded, collapsibleBodyId } = useCollapsible();
+  const { showNumberPrefix, expandedIndex } = useAccordion();
+  const { theme } = useTheme();
+
+  const toggleCollapse = (): void => onExpandChange(!isExpanded);
+  const isItemExpanded = expandedIndex === index;
+
+  const isPressed = useSharedValue(false);
+
+  const duration = castNativeType(getTransitionDuration(theme));
+  const easing = castNativeType(getTransitionEasing(theme));
+
+  const activeBackgroundColor = useSharedValue(
+    getBackgroundColor({ theme, isExpanded, isActive: true }),
+  );
+  const inActiveBackgroundColor = useSharedValue(
+    getBackgroundColor({ theme, isExpanded, isActive: false }),
+  );
+
+  /**
+   * `backgroundColor` is derived from `isExpanded` and pressed state.
+   *
+   * Since we can't directly access return values from JS functions (`getBackgroundColor`) on UI thread,
+   * this effect updates the shared values for active / inactive states which are used by `animatedStyles`.
+   */
+  useEffect(() => {
+    activeBackgroundColor.value = getBackgroundColor({ theme, isExpanded, isActive: true });
+    inActiveBackgroundColor.value = getBackgroundColor({ theme, isExpanded, isActive: false });
+  }, [isExpanded, activeBackgroundColor, inActiveBackgroundColor, theme]);
+
+  const animatedStyles = useAnimatedStyle(() => ({
+    backgroundColor: withTiming(
+      isPressed.value ? activeBackgroundColor.value : inActiveBackgroundColor.value,
+      {
+        duration,
+        easing,
+      },
+    ),
+  }));
+
+  const _showNumberPrefix = typeof index === 'number' && showNumberPrefix;
+
+  const _index = _showNumberPrefix ? (
+    <Heading size="small" marginRight="spacing.2">
+      {index + 1}.
+    </Heading>
+  ) : null;
+
+  const a11yLabel = _showNumberPrefix ? `${index + 1}. ${children}` : children;
+
+  const renderChildren: PressableProps['children'] = ({ pressed }) => {
+    isPressed.value = pressed;
+
+    const iconColor: IconProps['color'] = pressed
+      ? 'surface.action.icon.focus.lowContrast'
+      : 'surface.action.icon.default.lowContrast';
+
+    const _icon = Icon && (
+      <Icon size="medium" color={iconColor} marginRight="spacing.3" marginY="spacing.2" />
+    );
+
+    if (_index && _icon) {
+      throw new Error(`[Blade: Accordion]: showNumberPrefix and icon shouldn't be used together`);
+    }
+
+    return (
+      <BaseBox
+        display="flex"
+        flexDirection="row"
+        flex={1}
+        justifyContent="space-between"
+        alignItems="center"
+      >
+        <BaseBox
+          display="flex"
+          flexDirection="row"
+          alignItems="flex-start"
+          /**
+           * The `marginRight` is slightly larger here than web and the design spec (`spacing.4`),
```

Comment:
Should we change it in design as well in that case?

Reply:
I think should be fine, because this is for native only, that too for larger text. Generally by design guidelines the title text should be concise to fit in one line.